File: learning_tf_homework/src/tf_broadcaster.cpp
#include <ros/ros.h>
#include <tf/transform_broadcaster.h>

int main(int argc, char** argv) {
  ros::init(argc, argv, "robot_tf_broadcaster");

  ros::NodeHandle nh;

  ros::Rate loop_rate(100);

  tf::TransformBroadcaster br;

  while (nh.ok()) {
    tf::Transform tf_data;
    tf_data.setOrigin(tf::Vector3(0.1, 0.0, 0.2));

    br.sendTransform(tf::StampedTransform(tf_data, ros::Time::now(), "base_link", "base_laser"));

    loop_rate.sleep();
    ros::spinOnce();
  }

  return 0;
}

File: learning_tf_homework/scripts/tf_broadcaster.py
#! /usr/bin/env python3

import rospy
import tf

def main():
  rospy.init_node("robot_tf_broadcaster")
  
  tf_br = tf.TransformBroadcaster()

  rate = rospy.Rate(100)
  while not rospy.is_shutdown():
    translate_date = (0.1, 0.0, 0.2)
    rotation_data = (0, 0, 0, 1)
    tf_br.sendTransform(translate_date, rotation_data, rospy.Time.now(), "base_laser", "base_link")
    
    rate.sleep()

if __name__ == "__main__":
  main()
